ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessor.java

```diff
@@ -89,6 +89,15 @@ public boolean isSuccessful() {
 
   private final MiningBeneficiaryCalculator miningBeneficiaryCalculator;
 
+  public AbstractBlockProcessor(final AbstractBlockProcessor blockProcessor) {
```

Comment:
that would require exposing getters for the fields in AbstractBlockProcessor. I don't see how this can be done

Reply:
The abstract processor is used when you need a block processor that will change internal behaviour of processing a block (e.g. the block reward). What we need, is just a way to execute code before/after the actual block processing, without needing to know anything about what happens inside the `processBlock` method.

My suggestion would be changing `PrivacyBlockProcessor` to implement the `BlockProcessor` interface instead of extending the `AbstractBlockProcessor` abstract class. This would remove the need for the extra constructor in `AbstractBlockProcessor` and the need for the `rewardCoinbase` method override in `PrivacyBlockProcessor`.